pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -841,16 +1366,14 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject) in
 	if err != nil {
 		log.Fatal(err)
 	}
-	zedagentCtx.subBaseOsStatus = subBaseOsStatus
-	subBaseOsStatus.Activate()
 
-	subEdgeNodeCert, err := ps.NewSubscription(pubsub.SubscriptionOptions{
+	zedagentCtx.subEdgeNodeCert, err = ps.NewSubscription(pubsub.SubscriptionOptions{
 		AgentName:     "tpmmgr",
 		MyAgentName:   agentName,
 		TopicImpl:     types.EdgeNodeCert{},
 		Activate:      false,
```

Comment:
Why not set Activate true here as well?

Reply:
With persistent subscriptions (of which there are 4 in zedagent), it is necessary to split `Activate()` from `NewSubscription()`. This is because `Activate()` calls `populate()`, which calls `ModifyHandler` and there we typically expect the subscription handler (e.g. `zedagentCtx.subEdgeNodeCert`) to be already set in the context. So the assignment `zedagentCtx.subEdgeNodeCert, err = ...` must happen before activation. I noticed that we often put Activate as a separate call, but it is really only necessary for persistent subscriptions. I have added comment to each of these 4 zedagent subs with explanation.